hudi-hadoop-common/src/main/java/org/apache/parquet/avro/HoodieAvroReadSupport.java

```diff
@@ -51,6 +51,10 @@ public ReadContext init(Configuration configuration, Map<String, String> keyValu
       configuration.set(AvroWriteSupport.WRITE_OLD_LIST_STRUCTURE,
           "false", "support reading avro from non-legacy map/list in parquet file");
     }
+    if (legacyMode) {
```

Comment:
Is it possible to write a simple test so that the test fails before this fix and succeeds after the fix?

Reply:
Yes, struggling to do this with all Hudi test rails, but slowly proceeding.
I'm not sure exactly understand what you mean since if I have this code block it will anyway proceed.

Reply:
Done.
Added 10 potential scenarios,
marked those 4 that failed with the previous code and added comments around:
1. 2 are leading to exceptions we receive
2. 2 are leading to silent dataloss

Reply:
Added generation of those scenarios possible in theory(omitted some that require a lot of bootstrapping).
Also left separately 4 aforementioned scenarios, so the developer will be able to run them separately.

Reply:
Great.  What I meant is that new tests should be added, which failed before this fix and succeed after this fix, which is what you're doing.